Gemfile

```diff
@@ -156,3 +156,7 @@ gem 'connection_pool', require: false
 gem 'xorcist', '~> 1.1'
 
 gem 'cocoon', '~> 1.2'
+
+gem 'opentelemetry-sdk', require: false
+gem 'opentelemetry-exporter-otlp', require: false
+gem 'opentelemetry-instrumentation-all', require: false
```

Comment:
https://github.com/mastodon/mastodon/blob/main/Gemfile#L36-L38

It might be better to make it an optional installation like `devise_pam_authenticatable2`.

Reply:
🤔  Could maybe do that.

How does the optional `:pam_authentication` install get triggered today? I haven't yet read any instructions that include installing that bundler group (but I'm new here!) Does it happen at application runtime by [this line](https://github.com/mastodon/mastodon/blob/d4f973227c3a65833b6f8f34cab080900c77d4d6/config/application.rb#L51) in application.rb? Runtime dependency resolution and install (😱) is unsettling to me, personally.